### PR TITLE
feat: Add delete account feature

### PR DESCRIPTION
**Pull request by anhanh11001**
Fixes: #2114

Screenshots for the change:
ezgif-2-7c3e3618736e

### CODE REVIEW

File: app/src/main/res/layout/dialog_confirm_delete_account.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    <com.google.android.material.textfield.TextInputLayout
+        style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox.Dense"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:hint="@string/enter_account_password_message"
+        android:layout_marginBottom="@dimen/details_margin_small">
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:id="@+id/password"
```
Comment (ShridharGoel) [inline app/src/main/res/layout/dialog_confirm_delete_account.xml]: Add `android:inputType="textPassword"`

Comment (anhanh11001) [reply app/src/main/res/layout/dialog_confirm_delete_account.xml]: updated

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileViewModel.kt [hunk 2]
```diff
@@ -38,6 +45,43 @@ class ProfileViewModel(private val authService: AuthService, private val resourc
...
+        val query = """[{
+                |   'and':[{
+                |       'name':'is-event-online',
+                |       'op':'eq',
+                |       'val': 'true'
+                |    }]
+                |}]""".trimMargin().replace("'", "\"")
+
+        compositeDisposable += eventService.getEventsOfUser(authService.getId(), query)
+            .withDefaultSchedulers()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileViewModel.kt]: This is not your job. This is server's job

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileViewModel.kt]: I thought frontend also did the same. There is no endpoint to check whether an account is deletable or not, so we have to check whether there are active events attached to the account from client-side or not

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileViewModel.kt]: So should I just proceed to delete and then when server return delete fail, I'll display the message that there are events attached so the account is not deletable?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileViewModel.kt]: yes

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 3]
```diff
@@ -182,6 +204,92 @@ class ProfileFragment : Fragment(), BottomIconDoubleClick {
...
+    private fun confirmDeleteAccount() {
+        val progressDialog = progressDialog(context, getString(R.string.deleting_account))
+
+        loginViewModel.progress
+            .nonNull()
+            .observe(viewLifecycleOwner, Observer {
+                progressDialog.show(it)
+            })
+
+        loginViewModel.validPassword
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: Can you use chaining here, after password validation, directly use the data?

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: I don't think so, chaining is more about using result from the previous request for the second request, should not be nested

**Comment by anhanh11001** [comment]
Reopening as mistakenly closed

**Pull request merged**